array-map/map.js

```diff
@@ -0,0 +1,10 @@
+function map(array, transform) {
+  const mapArray = [];
+  for (let i = 0; i < array.length; i++) {
+    const result = transform(array[i], i, array);
+    if (result) {
+      mapArray.push(array[i]);
+    }
```

Comment:
For map you want to just push the `result` into the array, no need to check if it is truthy or not. You want to create a new array full of the "transformed" values.